pkg/network/http/ebpf_gotls.go

```diff
@@ -0,0 +1,212 @@
+		IncludeReturnLocations: false,
+		ParamLookupFunction:    lookup.GetWriteParams,
+	},
+	bininspect.ReadGoTLSFunc: {
+		IncludeReturnLocations: true,
+		ParamLookupFunction:    lookup.GetReadParams,
+	},
+	bininspect.CloseGoTLSFunc: {
+		IncludeReturnLocations: false,
+		ParamLookupFunction:    lookup.GetCloseParams,
+	},
+}
+
+var structFieldsLookupFunctions = map[bininspect.FieldIdentifier]bininspect.StructLookupFunction{
+	bininspect.StructOffsetTLSConn:     lookup.GetTLSConnInnerConnOffset,
+	bininspect.StructOffsetTCPConn:     lookup.GetTCPConnInnerConnOffset,
+	bininspect.StructOffsetNetConnFd:   lookup.GetConnFDOffset,
+	bininspect.StructOffsetNetFdPfd:    lookup.GetNetFD_PFDOffset,
+	bininspect.StructOffsetPollFdSysfd: lookup.GetFD_SysfdOffset,
+}
+
+type GoTLSProgram struct {
+	manager *manager.Manager
+}
+
+// Static evaluation to make sure we are not breaking the interface.
+var _ subprogram = &GoTLSProgram{}
+
+func NewGoTLSProgram(c *config.Config) (*GoTLSProgram, error) {
+	if !c.EnableHTTPSMonitoring {
+		return nil, nil
+	}
+	return &GoTLSProgram{}, nil
+}
+
+func (p *GoTLSProgram) ConfigureManager(m *manager.Manager) {
+	// TODO check if we support go TLS on the current arch
+	if p == nil {
+		return
+	}
+
+	p.manager = m
+	p.manager.Maps = append(p.manager.Maps, []*manager.Map{
+		{Name: probeDataMap},
+		{Name: readPartialCallsMap},
+	}...)
+	// Hooks will be added in runtime for each binary
+}
+
+func (p *GoTLSProgram) ConfigureOptions(options *manager.Options) {}
+
+func (p *GoTLSProgram) Start() {
+	if p == nil {
+		return
+	}
+	// In the future Start() should just initiate the new processes listener
+	// and this implementation should be done for each new process found.
+
+	binPath := os.Getenv("GO_TLS_TEST")
+	f, err := os.Open(binPath)
+	if err != nil {
+		log.Errorf("could not open file %q due to %w", binPath, err)
+		return
+	}
+	defer f.Close()
+	elfFile, err := elf.NewFile(f)
+	if err != nil {
+		log.Errorf("file %q could not be parsed as elf: %w", binPath, err)
+		return
+	}
+	result, err := bininspect.InspectNewProcessBinary(elfFile, functionsConfig, structFieldsLookupFunctions)
+
+	// result and bin path are being passed as parameters as a preparation for the future when we will have a process
+	// watcher, so we will run on more than one binary in one goTLSProgram.
+	p.addInspectionResultToMap(result, binPath)
+
+	p.attachHooks(result, binPath)
+
+}
+
+func (p *GoTLSProgram) addInspectionResultToMap(result *bininspect.Result, binPath string) {
+	probeData, err := inspectionResultToProbeData(result)
+	if err != nil {
+		log.Errorf("error while parsing inspection result: %w", err)
+		return
+	}
+	dataMap, _, err := p.manager.GetMap(probeDataMap)
+	if err != nil {
+		log.Errorf("%q map not found: %w", probeDataMap, err)
+		return
+	}
+
+	// Map key is the pid, so it will be identified in the probe as the relevant data
+	pidStr := strings.Split(binPath, "/")[2]
+	pid, err := strconv.ParseInt(pidStr, 10, 32)
```

Comment:
your'e right, but in the near future we won't have to make this assumptions (since we'll control the process watcher). For now I added docs & checking.

Reply:
please add a comment saying it is a temporary fix